### PR TITLE
fix(overflow-menu): increase width of the spacer to match button width

### LINKED ISSUE #3013: [StatefulTable] with row actions: when hovering over the row actions redundant white space over the row action button appears
### What package is this for?

- [x] React
- [ ] Angular

### To Reproduce

1. Go to [Stateful Table with Row actions](https://carbon-addons-iot-react.com/?path=/story/1-watson-iot-table-table--with-row-expansion-and-actions)
2. Click on three dots button (any row action)
3. Notice extra white space line appear over the bottom edge of the hovered button (please see screenshot)

image

### Environment/versions:

- OS: [e.g. MacOS, Windows]
- Browser: [e.g. chrome, safari]
- `carbon-addons-iot-react` version: [e.g. `v2.60.0`]

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #3013 #1885 

**Summary**

- The small spacer tab in the overflow menu wasn't quite long enough to fill the space leaving a small gap. This fixes that.

**Change List (commits, features, bugs, etc)**

- increase width of spacer tab to match button width

**Acceptance Test (how to verify the PR)**

- Go to [StatefulTable with row action](https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-statefultable--stateful-example-with-expansion-max-pages-and-column-resize) or [OverflowMenu stories](https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-overflowmenu--basic)
- open the overflow menu
- the spacer tab should match the width of the button

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Check other stories using overflow menu (in RTL) to confirm the fixes don't break anything
- https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--table-example-with-create-save-views
- https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--basic-table-with-full-row-edit-example
- https://deploy-preview-3031--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-overflowmenu--basic

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by bjornalm — APPROVED** [review]
Nice, work. This is not from your code changes, but it is related so perhaps we can fix it with this PR as well (or a new one of you want):

image

**Comment by kevinsperrine** [comment]
> Nice, work. This is not from your code changes, but it is related so perhaps we can fix it with this PR as well (or a new one of you want):
> 
> image

I actually think this is a carbon bug in this version, but I added a fix to my auto-positioning hook to account for it, and replaced the overflow menu in TableActionRow with the auto-positioning version (as requested in #1885), so I've re-requested a review for those changes.

**Comment by bjornalm** [comment]
Code looks good but the  build failed so I the regression test links point to the previous version

File: packages/react/src/hooks/usePopoverPositioning.jsx [hunk 1]
```diff
@@ -116,6 +118,12 @@ export const usePopoverPositioning = ({
         }
       }
 
+      let left = 0;
+
+      if (langDir === 'rtl' && isOverflowMenu) {
+        left = -15;
```
Comment (bjornalm) [inline packages/react/src/hooks/usePopoverPositioning.jsx:124]: Put this value 15 in a well named const that explains the origin?

**Review by bjornalm — APPROVED** [review]

**Comment by bjornalm** [comment]
I approved this, but I also found another related issue that also exists in next. 
image

File: packages/react/src/hooks/usePopoverPositioning.jsx (hunk 1 above)
Comment (kevinsperrine) [reply packages/react/src/hooks/usePopoverPositioning.jsx:124]: It's not necessary. Carbon has fixed it and I need to take it out in the carbon upgrade PR...

**Comment by kevinsperrine** [comment]
> I approved this, but I also found another related issue that also exists in next. image

Yeah, there's already an issue for this. https://github.com/carbon-design-system/carbon-addons-iot-react/issues/2951

File: packages/react/src/hooks/usePopoverPositioning.jsx (hunk 1 above)
Comment (kevinsperrine) [reply packages/react/src/hooks/usePopoverPositioning.jsx]: Actually, nvm. It's not. I'll add a constant.

**Review by davidicus — APPROVED** [review]

**Pull request merged**